src/app/pages/system/enclosure/components/views/enclosure-view/enclosures/enclosure-view/enclosure-view.directive.ts

```diff
@@ -19,45 +20,42 @@ export class EnclosureViewDirective implements AfterViewInit {
   @ViewChild('mySvg') private viewSvg: ElementRef<HTMLObjectElement>;
   protected svgPath: string;
   readonly enclosure = input.required<DashboardEnclosure>();
-  readonly viewOption = input.required<EnclosureSide>();
+  readonly enclosureSide = input.required<EnclosureSide>();
+  readonly enclosureView = input.required<EnclosureView>();
   private previousSelectRect: SVGRectElement;
 
   readonly viewSpecificSlots = computed(() => {
     const enclosure = this.enclosure();
-    const viewOption = this.viewOption();
+    const viewOption = this.enclosureSide();
     const allSlots = Object.entries(enclosure.elements['Array Device Slot']);
     const slots: Record<number, DashboardEnclosureSlot> = {};
-    let driveBayNumber = 1;
+
     switch (viewOption) {
       case EnclosureSide.Front:
         for (const slot of allSlots) {
           if (slot[1].is_front) {
-            slots[driveBayNumber] = { ...slot[1], drive_bay_number: +slot[0] };
-            driveBayNumber++;
+            slots[+slot[0]] = { ...slot[1], drive_bay_number: +slot[0] };
```

Comment:
I don't see what this code does now.

Reply:
I've seen `drive_bay_number` to be not assigned when returned from middleware. But don't have a system to check right now.